kernel/src/main/scala/iliad/Enum.scala

```diff
@@ -0,0 +1,23 @@
+package iliad
+
+import simulacrum.typeclass
+
+/** Represent discrete operations that can be performed on A */
+@typeclass trait Enum[A] {
+  def succ(x: A): A
+  def pred(x: A): A
+  def adj(x: A, y: A): Boolean = succ(x) == y
+}
+
+object Enum {
+  implicit val intEnum: Enum[Int] = new Enum[Int] {
+    override def succ(x: Int): Int = x + 1
+    override def pred(x: Int): Int = x - 1
+  }
+
```

Comment:
Not fond of having the standard ones inside here rather than in the `int` object instance.